docs/framework/api-architecture.md

```diff
@@ -0,0 +1,175 @@
+# API architecture
+
+
+The term 'API' refers to that code stored in the top level api folder.
```

Comment:
I'm not sure what you mean by *"Is this brought over?" This confusion leaves me unsure how to address your comment, so I'll be ignoring it in pursuit of merging this PR. Happy to make other changes after merging.